Lib/ast.py

```diff
@@ -633,14 +691,15 @@ def visit_Module(self, node):
 
     def visit_Expr(self, node):
         self.fill()
+        self.set_precedence("PR_TEST", node.value)
         self.traverse(node.value)
 
     def visit_NamedExpr(self, node):
-        self.write("(")
-        self.traverse(node.target)
-        self.write(" := ")
-        self.traverse(node.value)
-        self.write(")")
+        with self.delimit(self.check_precedence("PR_TUPLE", node)):
```

Comment:
To make the PR more readable, I suggest to write first PR which only added delim() context manager but a version without argument. A PR basically only to change the indentation. Once it's merged, rebased this PR on top of it.

A PR just to add "(" an ")" as context manager sounds like a good idea to me, even if this PR is not accepted.